src/py42/services/storage/restore.py

```diff
@@ -69,5 +77,23 @@ def start_push_restore(
             u"permitRestoreToDifferentOsVersion": permit_restore_to_different_os_version,
             u"filePermissions": file_permissions,
             u"restoreFullPath": restore_full_path,
+            u"fileLocation": file_location,
         }
-        return self._connection.post(uri, json=json_dict)
+        try:
+            return self._connection.post(uri, json=json_dict)
+        except Py42BadRequestError as err:
+            if u"CREATE_FAILED" in err.response.text:
+                additional_message = None
+                if (
+                    device_guid != accepting_device_guid
+                    and file_location == PushRestoreLocation.ORIGINAL
+                ):
+                    additional_message = (
+                        u"Warning: Trying to restore to the original "
+                        "location when the accepting GUID '{}' is "
+                        "different from the archive source GUID "
```

Comment:
Can this be the cause of the CREATE_FAILED error, but the server response doesn't make it clear that's why it failed?

Reply:
We are not sure that it is the cause.

However, we are sure that we got the pre-condition for the cause and that the GUIDs are different, thus, we are adding an additional WARNING to the error message.  

However, do you think that is too assumptious?

Reply:
Or is this obviously the reason of the cause. I don't understand 100% all the use-cases for push restore, but I know tht the `ORIGINAL_LOCATION` strategy can have to do with device migration, in which case the device guids would already be the same one as they would be restoring a device to the new one (putting the files there for the first time).

Reply:
The server really does not make it super clear, just says the restore either does not exist or couldn't be created.

Reply:
Yeah, I'm not exactly sure how things can break in this specific case, which is why I was asking. 

If you try ORIGINAL_LOCATION with different source/target guids, and it fails. Then switch to TARGET_DIRECTORY and it works with only that change, then it makes sense to have this error text. 

But I wouldn't want to send someone down a rabbit-trail with this message if that isn't actually a problem. 

Does the UI present similar warnings at all? I think there's one about cross-operating system restores (.e.g a Windows archive being pushed to a Mac/Linux device).

Reply:
Yes, that is exactly what I tried - took a working restore, that set the restore_path and everything, and it was using two devices... and then I switched to ORIGINAL_LOCATION and I got this generic error about it not being found or created.

As far as permitting different OS restores, I haven't tried messing around with that parameter at all - do you think that is needed now?

Reply:
Ok, if we know that just changing that flag can cause it to fail, then the warning is warranted as is. I don't think we need to do anything specific with cross-OS restores, it's very edge case.

Reply:
Hmm... I am wondering if `restoreFullPath` is set True, then I could see this making sense, so maybe it's best not to have this additional warning.

Reply:
Yeah I removed it for now